zap/src/test/java/org/parosproxy/paros/network/HttpRequestHeaderUnitTest.java

```diff
@@ -240,6 +240,48 @@ void isCssShouldReturnTrueWhenUrlIndicatesCss(String url) {
         assertTrue(reqHeader.isCss());
     }
 
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                "http://example.org/images/file.ext", // In directory path
+                "http://xrayimage.com/", // In domain name
+                "https://example.image", // In domain extension (TLD)
+                "https://example.image/dir/file.ext", // In domain extension (TLD)
+                "https://example.gif/dir/file.ext", // In domain extension (TLD)
+                "https://example.org/dir/file?foo=bar&thing=image", // In parameter value
+                "https://example.org/dir/file?foo=bar&thing=gif", // In parameter value
+                "http://example.org/image/file.ext?foo=bar&type=.image", // In parameter value
+                // including period
+                "http://example.org/image/file.ext?foo=bar&type=.gif", // In parameter value
+                // including period
+                "http://example.org/image/file.ext?foo=bar&thing=image.gif", // In parameter value,
+                // plausible filename
+                "https://example.org/dir/file?foo=bar&image=file.ext", // In parameter name
+                "https://example.org/dir/file?foo=bar&image=file.gif" // In parameter name and value
+            })
+    void isImageShouldReturnFalseWhenUrlDoesNotIndicateImage(String url) {
+        // Given
+        HttpRequestHeader reqHeader = createRequestHeader(url);
+        // When / Then
+        assertFalse(reqHeader.isImage());
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                "http://example.org/example.svg", // In path
+                "http://example.org/assets/images/example.gif", // In deeper path
+                "http://example.org/images/example.png?foo=bar", // In path, ignoring params
+                "http://example.org/images/example.jpg?foo=bar&thing=.jpg", // In path, ignoring
+                // params
+            })
+    void isImageShouldReturnTrueWhenUrlIndicatesImage(String url) {
```

Comment:
Ideally this should check all extensions not just some in some places.

Reply:
I'll see what i can come up with.